lib/server/db/model/user.js

```diff
@@ -1,14 +1,33 @@
-var mongoose = require('mongoose');
+var mongoose  = require('mongoose'),
+    validator = require('node-mongoose-validator');
+
 mongoose.connect('mongodb://localhost/beans');
 
-var User = mongoose.model(
-    'User',
-    {
-        username: String,
-        password: String,
-        friends: []
-    }
-);
+// define a schema
+var userSchema = new mongoose.Schema({
+    firstName: { type: String, required: true },
+    lastName:  { type: String, required: true },
+    username:  { type: String, required: true },
+    password:  { type: String, required: true }
+
+});
+
+// Validations
+userSchema.path('firstName').validate(validator.notEmpty(), 'Please provide a first name.');
+userSchema.path('lastName').validate(validator.notEmpty(), 'Please provide a last name.');
+userSchema.path('username').validate(validator.notEmpty(), 'Please provide a username.');
+userSchema.path('password').validate(validator.notEmpty(), 'Please provide a password.');
+
+// compile our model
+var User = mongoose.model("User", userSchema);
+
+// Virtual property getter & setter for user.fullName
+userSchema.virtual('fullName').
+    get(function () { return this.firstName + ' ' + this.lastName; }).
```

Comment:
When chaining methods together, JavaScript standard is to place the dot on the same line as the chained method:

```
userSchema.virtual("fullName")
     .get()
     .set();
```